File: app/src/androidTest/java/djordjeh/architecture/mvp/dagger/TestApplicationComponent.java
package djordjeh.architecture.mvp.dagger;


import javax.inject.Singleton;

import dagger.Component;
import dagger.android.AndroidInjector;
import dagger.android.support.DaggerApplication;

@Singleton
@Component(modules = {
        ActivityBindingModule.class,
        TestApplicationModule.class
})
public interface TestApplicationComponent extends AndroidInjector<DaggerApplication> {

    @Override
    void inject(DaggerApplication application);
}

File: app/src/androidTest/java/djordjeh/architecture/mvp/dagger/TestApplicationComponent.kt
package djordjeh.architecture.mvp.dagger


import javax.inject.Singleton

import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.DaggerApplication

@Singleton
@Component(modules = [ActivityBindingModule::class, TestApplicationModule::class])
interface TestApplicationComponent : AndroidInjector<DaggerApplication> {

    override fun inject(application: DaggerApplication)
}
